packages/amplify-cli/src/extensions/amplify-helpers/get-root-stack-id.ts

```diff
@@ -4,8 +4,8 @@ export function getRootStackId(): string {
   const teamProviderInfo = stateManager.getTeamProviderInfo();
   const { envName } = stateManager.getLocalEnvInfo();
   const envTeamProviderInfo = teamProviderInfo[envName];
-  if (envTeamProviderInfo && envTeamProviderInfo.awscloudformation) {
-    const stackId = envTeamProviderInfo.awscloudformation.StackId;
+  const stackId = envTeamProviderInfo?.awscloudformation?.StackId;
+  if (typeof stackId === 'string') {
```

Comment:
What else could it be if not undefined? why checking for type here?

Reply:
It should only ever be a string or undefined but checking for string is a stronger assertion than checking for not undefined